core/src/main/scala/cats/MonadDefer.scala

```diff
@@ -0,0 +1,21 @@
+package cats
+
+import simulacrum.typeclass
+import cats.Eval.always
+
+/**
+ * A [[Monad monad]] that allows for arbitrarily delaying the
+ * evaluation of an operation, triggering its execution on each run.
+ *
+ * @see [[ApplicativeEval]] for capturing effects in an `F[_]`
+ *     applicative context, but without the repeating side-effects
+ *     requirement.
+ */
+@typeclass trait MonadDefer[F[_]] extends Monad[F] with ApplicativeEval[F] {
```

Comment:
Is it worth providing a default instance of `MonadDefer` implemented as `pure(()).map(_ => a)` and `pure(()).flatMap(_ => fa)` for any `Monad` `M`? This way we could have a `MonadDefer` instance even for `Id` which can be useful.